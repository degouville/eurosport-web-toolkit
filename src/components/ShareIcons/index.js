import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';
import ShareSvg from 'src/assets/share.svg';
import ShareIcon from '../../elements/ShareIcon';
import { frenchGray } from '../../colors';
import { H5 } from '../../typography';

const Wrapper = styled.div`
  width: 320px;
  display: flex;
  align-items: center;
`;

const Icons = styled(ShareIcon)`
  margin-right: 20px;
`;

const Label = styled(H5)`
  text-transform: uppercase;
  color: ${frenchGray};
  margin-right: 20px;
  margin-top: 1px;
`;

const Share = styled.div`
  background: url(${ShareSvg}) no-repeat;
  width: 11px;
  height: 11px;
  margin-right: 10px;
`;

const ShareIcons = ({ label, icons, labelIcon }) => (
  <Wrapper>
    {labelIcon && <Share />}
    {label && <Label>{label}</Label>}
    {icons && icons.map(icon => <Icons key={icon.icon} icon={icon.icon} size={icon.size} onClick={icon.onClick} />)}
  </Wrapper>
);

ShareIcons.propTypes = {
  icons: PropTypes.arrayOf(
    PropTypes.shape({
      icon: PropTypes.string,
      size: PropTypes.number,
      onClick: PropTypes.func,
    })
  ).isRequired,
  label: PropTypes.string,
  labelIcon: PropTypes.bool,
};

ShareIcons.defaultProps = {
  label: null,
  labelIcon: false,
};

export default ShareIcons;
