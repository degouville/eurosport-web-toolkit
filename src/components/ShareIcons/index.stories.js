import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import ShareIcons from './index';

const ShareIconsStories = storiesOf('Components|ShareIcons', module);

const icons = [
  { icon: 'facebook', size: 32, onClick: action('facebook') },
  { icon: 'twitter', size: 32, onClick: action('twitter') },
  { icon: 'whatsapp', size: 32, onClick: action('whatsapp') },
];

ShareIconsStories.add('ShareModule', () => (
  <ShareIcons labelIcon={boolean('Label Icon', true)} icons={icons} label={text('Share Label', 'share')} />
));
