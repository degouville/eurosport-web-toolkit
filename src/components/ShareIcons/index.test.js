import React from 'react';
import { shallow } from 'enzyme';
import ShareIcons from '.';

const icons = [
  { icon: 'facebook', size: 32, onClick: () => {} },
  { icon: 'twitter', size: 32, onClick: () => {} },
  { icon: 'whatsapp', size: 32, onClick: () => {} },
];

describe('Footer', () => {
  it('renders when everything passed in', () => {
    expect(shallow(<ShareIcons labelIcon icons={icons} label="share" />)).toMatchSnapshot();
  });

  it('renders without label', () => {
    expect(shallow(<ShareIcons labelIcon icons={icons} />)).toMatchSnapshot();
  });

  it('renders with no label share icon', () => {
    expect(shallow(<ShareIcons icons={icons} />)).toMatchSnapshot();
  });
});
