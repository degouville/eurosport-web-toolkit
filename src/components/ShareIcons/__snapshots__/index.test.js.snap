// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders when everything passed in 1`] = `
<Styled(div)>
  <Styled(div) />
  <Styled(h5)>
    share
  </Styled(h5)>
  <Styled(ShareIcon)
    icon="facebook"
    key="facebook"
    onClick={[Function]}
    size={32}
  />
  <Styled(ShareIcon)
    icon="twitter"
    key="twitter"
    onClick={[Function]}
    size={32}
  />
  <Styled(ShareIcon)
    icon="whatsapp"
    key="whatsapp"
    onClick={[Function]}
    size={32}
  />
</Styled(div)>
`;

exports[`Footer renders with no label share icon 1`] = `
<Styled(div)>
  <Styled(ShareIcon)
    icon="facebook"
    key="facebook"
    onClick={[Function]}
    size={32}
  />
  <Styled(ShareIcon)
    icon="twitter"
    key="twitter"
    onClick={[Function]}
    size={32}
  />
  <Styled(ShareIcon)
    icon="whatsapp"
    key="whatsapp"
    onClick={[Function]}
    size={32}
  />
</Styled(div)>
`;

exports[`Footer renders without label 1`] = `
<Styled(div)>
  <Styled(div) />
  <Styled(ShareIcon)
    icon="facebook"
    key="facebook"
    onClick={[Function]}
    size={32}
  />
  <Styled(ShareIcon)
    icon="twitter"
    key="twitter"
    onClick={[Function]}
    size={32}
  />
  <Styled(ShareIcon)
    icon="whatsapp"
    key="whatsapp"
    onClick={[Function]}
    size={32}
  />
</Styled(div)>
`;
